v2/streaming-data-generator/src/test/java/com/google/cloud/teleport/v2/templates/StreamingDataGeneratorPerformanceIT.java

```diff
@@ -200,11 +210,74 @@ public void testGenerateBigQuery10gb() throws IOException, ParseException, Inter
     JobInfo info = dataflowClient.launch(PROJECT, REGION, options);
     assertThat(info.state()).isIn(JobState.ACTIVE_STATES);
     Result result = dataflowOperator.waitUntilDone(createConfig(info, Duration.ofMinutes(30)));
+
     // Assert
     assertThat(result).isEqualTo(Result.JOB_FINISHED);
     assertThat(bigQueryResourceManager.readTable(jobName).getTotalRows()).isGreaterThan(0);
 
     // export results
     exportMetricsToBigQuery(info, getMetrics(info, FAKE_DATA_PCOLLECTION));
   }
+
+  @Test
+  public void testGenerateSpanner10gb() throws IOException, ParseException, InterruptedException {
+    // Set up resource manager
+    spannerResourceManager =
+        DefaultSpannerResourceManager.builder(testName.getMethodName(), PROJECT, REGION).build();
+    String createTableStatement =
+        String.format(
+            "CREATE TABLE `%s` (\n"
+                + "  eventId STRING(1024) NOT NULL,\n"
+                + "  eventTimestamp INT64,\n"
+                + "  ipv4 STRING(1024),\n"
+                + "  ipv6 STRING(1024),\n"
+                + "  country STRING(1024),\n"
+                + "  username STRING(1024),\n"
+                + "  quest STRING(1024),\n"
+                + "  score INT64,\n"
+                + "  completed BOOL,\n"
+                + ") PRIMARY KEY(eventId)",
+            testName.getMethodName());
+    ImmutableList<String> columnNames =
+        ImmutableList.of(
+            "eventId",
+            "eventTimestamp",
+            "ipv4",
+            "ipv6",
+            "country",
+            "username",
+            "quest",
+            "score",
+            "completed");
+    spannerResourceManager.createTable(createTableStatement);
+    // Arrange
+    String name = testName.getMethodName();
+    String jobName = createJobName(name);
+    LaunchConfig options =
+        LaunchConfig.builder(jobName, SPEC_PATH)
+            .addParameter("schemaTemplate", SchemaTemplate.GAME_EVENT.name())
+            .addParameter("qps", "1000000")
+            .addParameter("messagesLimit", NUM_MESSAGES)
+            .addParameter("sinkType", "SPANNER")
+            .addParameter("projectId", PROJECT)
+            .addParameter("spannerInstanceName", spannerResourceManager.getInstanceId())
+            .addParameter("spannerDatabaseName", spannerResourceManager.getDatabaseId())
+            .addParameter("spannerTableName", testName.getMethodName())
+            .addParameter("numWorkers", "50")
+            .addParameter("maxNumWorkers", "100")
```

Comment:
just curious - would somehow externalize QPS/numWorkers/maxNumWorkers help? or use parameterized tests to have different sets of tests?

Reply:
Having parameterized tests for QPS/num workers/max num workers won't give us much information, but will increase our tests by a lot. Also, most of the time the pipeline autoscales down/up in a matter of a few minutes.